segment.go

```diff
@@ -89,19 +98,21 @@ func (*ZapPlugin) Open(path string) (segment.Segment, error) {
 // SegmentBase is a memory only, read-only implementation of the
 // segment.Segment interface, using zap's data representation.
 type SegmentBase struct {
-	mem               []byte
-	memCRC            uint32
-	chunkMode         uint32
-	fieldsMap         map[string]uint16 // fieldName -> fieldID+1
-	fieldsInv         []string          // fieldID -> fieldName
-	numDocs           uint64
-	storedIndexOffset uint64
-	fieldsIndexOffset uint64
-	docValueOffset    uint64
-	dictLocs          []uint64
-	fieldDvReaders    map[uint16]*docValueReader // naive chunk cache per field
-	fieldDvNames      []string                   // field names cached in fieldDvReaders
-	size              uint64
+	mem                  []byte
+	memCRC               uint32
+	chunkMode            uint32
+	fieldsMap            map[string]uint16   // fieldName -> fieldID+1
+	fieldsInv            []string            // fieldID -> fieldName
+	fieldsSectionsMap    []map[uint16]uint64 // fieldID -> section -> address
+	numDocs              uint64
+	storedIndexOffset    uint64
+	fieldsIndexOffset    uint64
+	newFieldsIndexOffset uint64
```

Comment:
I suppose the reason you kept `newFieldsIndexOffset` along side `fieldsIndexOffset` is for backward compatibility? The new attribute could use a better name implying sections are in play.

Reply:
it was more related to the fact that the newFieldsIndexOffset relates to information that has fieldsIndex info tagged alongside the section type informations.